src/main/scala/is/hail/asm4s/StagedBitSet.scala

```diff
@@ -0,0 +1,37 @@
+package is.hail.asm4s
+
+import org.objectweb.asm.tree._
+import scala.collection.generic.Growable
+
+class StagedBitSet(fb: FunctionBuilder[_]) {
+  private var used = 0
+  private var bits: LocalRef[Long] = null
+  private var count = 0
+
+  def newBit(): SettableBit = {
+    if (used >= 64 || bits == null) {
+      bits = fb.newLocal[Long]
+      count += 1
+      fb.emit(bits.store(0L))
```

Comment:
I'm going to approve this, but this seems lightly dangerous, although I'm not sure what else to do, except maybe have `newBit` return a pair including an initialization code.  Problem I'm imagining:

```
  emit if
    val mx = sbs.newBit() # first bit in word
    ...
  end if
  val my = sbs.newBit() // second bit in long, uninitialized!
```